pg_search/src/query/mod.rs

```diff
@@ -637,9 +683,59 @@ fn check_range_bounds(
 
     let upper_bound = match (typeoid, upper_bound.clone()) {
         // Included U64 needs to be canonicalized
-        (_, Bound::Included(OwnedValue::U64(n))) => Bound::Excluded(OwnedValue::U64(n + 1)),
+        (_, Bound::Included(OwnedValue::U64(n))) => {
+            // Determine if this is for a signed integer field (int2, int4, int8)
+            let is_signed_int_field = matches!(
+                typeoid,
+                PgOid::BuiltIn(
+                    PgBuiltInOids::INT2OID | PgBuiltInOids::INT4OID | PgBuiltInOids::INT8OID
+                )
+            );
+
+            if is_signed_int_field {
+                // For signed integer fields, check against i64::MAX
+                let i64_max = i64::MAX as u64;
+
+                if n > i64_max {
+                    // Value exceeds i64 range - this shouldn't match any i64 values
+                    Bound::Unbounded
+                } else if n == i64_max {
+                    // At boundary: convert to I64 and keep as Included (don't canonicalize)
+                    // Incrementing would overflow, so keep as-is to avoid type mismatch
+                    Bound::Included(OwnedValue::I64(i64::MAX))
+                } else {
+                    // Safe to increment within i64 range
+                    match n.checked_add(1) {
+                        Some(next) if next <= i64_max => Bound::Excluded(OwnedValue::U64(next)),
+                        _ => {
+                            // Shouldn't reach here since n < i64_max
+                            Bound::Unbounded
+                        }
```

Comment:
Should use `unreachable!("Shouldn't reach here, since n < i16_max")` for something like this probably.